Update bicep references (#1149)
radius-project/docs: docs/content/guides/tooling/bicepconfig/overview/index.md

description: "Setup the Bicep configuration file to author and deploy Radius-types"
weight: 100
categories: "Overview"
tags: ["Bicep"]
---

To use the features provided by the official Bicep compiler with Radius, certain configurations need to be defined. These are defined in a `bicepconfig.json` file that lives in your application's directory. 

## What is a `bicepconfig.json`?

The `bicepconfig.json` allows Bicep compiler to consume and use Radius-types stored in an OCI registry. There are two extensions that are enabled by default in the `bicepconfig.json` so that you can use Radius and AWS resources. The "radius" extension contains the schema information for all Radius-maintained resources, and the "aws" extension contains the schema information for AWS resources. You can optionally add any other settings that are relevant to your application. There are two ways to generate a `bicepconfig.json` with Radius. 

## Automatically generate a `bicepconfig.json` via `rad init`

{{< read file= "/shared-content/installation/bicepconfig/rad-init.md" >}}

## Manually create a `bicepconfig.json` 

{{< read file= "/shared-content/installation/bicepconfig/manual.md" >}}

> ## Author and deploy Radius-types

Author: 👍

> {{< alert title="Replace import statements with extension" color="warning" >}} Radius is now merged with the official Bicep. If you have bicep files with `import radius as radius` statements please replace them with `extension radius` to use Radius-types .

Author: ```suggestion
{{< alert title="Replace import statements with extension" color="warning" >}} Radius is now merged with the official Bicep. If you have bicep files with the following import statements, please replace them with as needed.

 1. `import radius as radius` should become `extension radius` to use Radius types 
 1. `import aws as aws` should become `extension aws` to use AWS types
 1. `import kubernetes as kubernetes {}` should become `extension kubernetes with {} as kubernetes` to use Kubernetes types
```

---
type: docs
title: "Overview: Bicep configuration"
linkTitle: "Overview"
description: "Setup the Bicep configuration file to author and deploy Radius-types"
weight: 100
categories: "Overview"
tags: ["Bicep"]
---

To use the features provided by the official Bicep compiler with Radius, certain configurations need to be defined. These are defined in a `bicepconfig.json` file that lives in your application's directory. 

## What is a `bicepconfig.json`?

The `bicepconfig.json` allows Bicep compiler to consume and use Radius-types stored in an OCI registry. There are two extensions that are enabled by default in the `bicepconfig.json` so that you can use Radius and AWS resources. The "radius" extension contains the schema information for all Radius-maintained resources, and the "aws" extension contains the schema information for AWS resources. You can optionally add any other settings that are relevant to your application. There are two ways to generate a `bicepconfig.json` with Radius. 

> 

Reviewer: ```suggestion
```

weight: 100
categories: "Overview"
tags: ["Bicep"]
---

To use the features provided by the official Bicep compiler with Radius, certain configurations need to be defined. These are defined in a `bicepconfig.json` file that lives in your application's directory. 

## What is a `bicepconfig.json`?

The `bicepconfig.json` allows the Bicep compiler to consume and use Radius-types stored in an OCI registry. There are two extensions that are enabled by default in the `bicepconfig.json` so that you can use Radius and AWS resources. The "radius" extension contains the schema information for all Radius-maintained resources, and the "aws" extension contains the schema information for AWS resources. You can optionally add any other settings that are relevant to your application. There are two ways to generate a `bicepconfig.json` with Radius. 

## Automatically generate a `bicepconfig.json` via `rad init`

{{< read file= "/shared-content/installation/bicepconfig/rad-init.md" >}}

## Manually create a `bicepconfig.json` 

{{< read file= "/shared-content/installation/bicepconfig/manual.md" >}}

## Author and deploy Radius-types

> {{< alert title="Replace import statements with extension" color="warning" >}} Radius is now merged with the official Bicep. If you have bicep files with the following import statements, please replace them with as needed.

Reviewer: ```suggestion
{{< alert title="Replace import statements with extension" color="warning" >}} Radius is now merged with the official Bicep. If you have bicep files with the following import statements, please replace them as needed.
```

---
type: docs

> title: "Overview: Bicep configuration file"

Reviewer: ```suggestion
title: "Overview: Bicep configuration"
```

linkTitle: "Overview"

> description: "Setup the Bicep configuration file with Radius-managed types"

Reviewer: ```suggestion
description: "Setup the Bicep configuration file to author and deploy Radius-managed types"
```

weight: 100
categories: "Overview"

> tags: ["Bicep", "bicepconfig"]

Reviewer: ```suggestion
tags: ["Bicep"]
```

Reviewer: less tags are better to manage the tag cloud in the homepage

---

In order to use the features provided by the official Bicep compiler with Radius, certain configurations need to be defined. These are defined in a `bicepconfig.json` file that lives in your application's directory. There are two ways to generate a `bicepconfig.json` with Radius. 

> ## Option 1: Generate a `bicepconfig.json` using Radius

Reviewer: ```suggestion
## Option 1: Automatically generate a `bicepconfig.json` via `rad init`
```

1. Install Radius CLI

{{< read file= "/shared-content/installation/rad-cli/install-rad-cli.md" >}}

1. Create a new directory for your app and navigate into it:

```bash
mkdir first-app
cd first-app
```

> Initialize Radius. Select `Yes` when asked to setup application in the current directory. This will automatically generate a `bicepconfig.json` with the correct setup in your application's directory. 

Reviewer: ```suggestion
1. Initialize Radius. Select `Yes` when asked to setup application in the current directory.  
```

```bash
rad init
```

> 

Reviewer: ```suggestion

This will automatically generate a `bicepconfig.json` with the correct setup in your application's directory.

```

## Option 2: Manually create a `bicepconfig.json` 

1. Create a `bicepconfig.json` in your application's directory. `release-version` should correspond to the current release version in the form of `major.minor` (e.g. `0.36`). 

```json
{
	"experimentalFeaturesEnabled": {
		"extensibility": true,
		"extensionRegistry": true,
		"dynamicTypeLoading": true
	},
	"extensions": {
		"radius": "br:biceptypes.azurecr.io/radius:<release-version>",
		"aws": "br:biceptypes.azurecr.io/aws:<release-version>"
	}
}
```

> A `bicepconfig.json` file is required to use Radius with the official Bicep compiler but you can optionally add any other settings that are relevant to your application. These configurations allow Bicep to consume and use Radius-managed types stored in an OCI registry. There are two extensions that are enabled by default in the `bicepconfig.json` so that you can use Radius and AWS resources. The "radius" extension contains the schema information for all Radius-maintained resources, and the "aws" extension contains the schema information for AWS resources. 

Reviewer: ```suggestion
These configurations allow Bicep to consume and use Radius-managed types stored in an OCI registry. There are two extensions that are enabled by default in the `bicepconfig.json` so that you can use Radius and AWS resources. The "radius" extension contains the schema information for all Radius-maintained resources, and the "aws" extension contains the schema information for AWS resources. You can optionally add any other settings that are relevant to your application.
```
